src/mlpack/methods/ann/layer/nearest_interpolation_impl.hpp

```diff
@@ -0,0 +1,228 @@
+  depth(0),
+  batchSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+NearestInterpolationType<MatType>::
+NearestInterpolationType(const size_t inRowSize,
+                         const size_t inColSize,
+                         const size_t outRowSize,
+                         const size_t outColSize,
+                         const size_t depth) :
+  Layer<MatType>(),
+  inRowSize(inRowSize),
+  inColSize(inColSize),
+  outRowSize(outRowSize),
+  outColSize(outColSize),
+  depth(depth),
+  batchSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+NearestInterpolationType<MatType>::
+NearestInterpolationType(const NearestInterpolationType& other) :
+  Layer<MatType>(),
+  inRowSize(other.inRowSize),
+  inColSize(other.inColSize),
+  outRowSize(other.outRowSize),
+  outColSize(other.outColSize),
+  depth(other.depth),
+  batchSize(other.batchSize)
+{
+  // Nothing to do here. 
+}
+
+template<typename MatType>
+NearestInterpolationType<MatType>::
+NearestInterpolationType(NearestInterpolationType&& other) :
+  Layer<MatType>(std::move(other)),
+  inRowSize(std::move(other.inRowSize)),
+  inColSize(std::move(other.inColSize)),
+  outRowSize(std::move(other.outRowSize)),
+  outColSize(std::move(other.outColSize)),
+  depth(std::move(other.depth)),
+  batchSize(std::move(other.batchSize))
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+NearestInterpolationType<MatType>&
+NearestInterpolationType<MatType>::
+operator=(const NearestInterpolationType& other) 
+{
+  if (&other != this) {
+    Layer<MatType>::operator=(other);
+    inRowSize = other.inRowSize;
+    inColSize = other.inColSize;
+    outRowSize = other.outRowSize;
+    outColSize = other.outColSize;
+    depth = other.depth;
+    batchSize = other.batchSize;
+  }
+  return *this;
+}
+
+template<typename MatType>
+NearestInterpolationType<MatType>&
+NearestInterpolationType<MatType>::
+operator=(NearestInterpolationType&& other) 
+{
+  if (&other != this) {
+    Layer<MatType>::operator=(std::move(other));
+    inRowSize = std::move(other.inRowSize);
+    inColSize = std::move(other.inColSize);
+    outRowSize = std::move(other.outRowSize);
+    outColSize = std::move(other.outColSize);
+    depth = std::move(other.depth);
+    batchSize = std::move(other.batchSize);
+  }
+  return *this;
+}
+
+template<typename MatType>
+void NearestInterpolationType<MatType>::Forward(
+  const MatType& input, MatType& output)
+{
+  batchSize = input.n_cols;
+  if (output.is_empty())
+    output.set_size(outRowSize * outColSize * depth, batchSize);
+  else
+  {
+    assert(output.n_rows == outRowSize * outColSize * depth);
+    assert(output.n_cols == batchSize);
+  }
+
+  assert(inRowSize >= 2);
+  assert(inColSize >= 2);
+
+  arma::cube inputAsCube;
+  arma::cube outputAsCube;
+
+  MakeAlias(inputAsCube, input, inRowSize, inColSize, depth*batchSize, 0, false);
+  MakeAlias(outputAsCube, output, outRowSize, outColSize, depth*batchSize, 0, true);
```

Comment:
```suggestion
  MakeAlias(inputAsCube, input, inRowSize, inColSize, depth * batchSize, 0, false);
  MakeAlias(outputAsCube, output, outRowSize, outColSize, depth * batchSize, 0, true);
```

Reply:
These lines are also a little over 80 characters so if you can wrap them (the same in `Backward()`) it would be great 😄